Pipeline/CreateUCDComponentVersion/README.md

```diff
@@ -1,69 +1,243 @@
-# IBM UCD Packaging based on IBM DBB Build Report
+# IBM UCD Packaging Based on IBM DBB Build Report
 
-An important step in the pipeline is to generate a deployable package. This sample groovy script
-- extracts information about the build outputs from the DBB BuildReport.json
-- generates the UCD shiplist ```shiplist.xml``` file
-- invokes the ```buztool.sh``` of UCD
+## Summary
+
+An important step in the pipeline is to generate a deployable package. This sample Groovy script:
+
+- Extracts information about the build outputs from the Dependency Based Build (DBB) `BuildReport.json`. The script is able to take a single DBB build report or multiple build reports to build a cumulative package across multiple incremental builds. 
+- Generates the UrbanCode Deploy (UCD) shiplist `shiplist.xml` file.
+- Invokes the `buztool.sh` with the appropriate configuration to store the binary package in the artifact repository and to register a new UCD component version.
+
+## High-level Processing Flow
+
+This section provides a more detailed explanation of how the CreateUCDComponentVersion script works and what it does.
+
+1. **Initialization**
+   1. Read command line parameters.
+   1. Read any application and global properties files that are passed as a comma-separated list via `--packagingPropFiles`.
+       (This parameter is optional for UCD package format v1, but required when using UCD package format v2 - see `--ucdV2PackageFormat`.)
+
+1. **Process the DBB build report(s)**
+   1. Either read DBB's `BuildReport.json` from the pipeline work directory, or loop through the list of provided DBB build reports (using the `--buildReportOrder` or `--buildReportOrderFile` option).
+   1. Parse and extract build output information for records of type *ExecuteRecord* and *CopyToPDSRecord*. (Requires at least DBB 1.0.8.)
+   1. Parse and extract the build output information for deleted build outputs of type *Delete_Record* written to the build report by zAppBuild. (Requires at least DBB 1.1.3, as the script uses the AnyTypeRecord API introduced in this version.)
+
+1. **Generate the UCD `shiplist.xml` file and invoke the UCD packaging step**
+   1. Generate the container records for the UCD shiplist for build outputs in partitioned datasets. (For more details, see the [IBM Docs UCD Shiplist](https://www.ibm.com/docs/en/urbancode-deploy/7.2.2?topic=SS4GSP_7.2.2/com.ibm.udeploy.doc/topics/zos_shiplistfiles.html) page.)
```

Comment:
That sentence sounds a bit complex. Maybe something like "Generate the UCD shiplist records related to build outputs in partitioned datasets"

Reply:
I know it's the UCD terminology, but we may need to simplify it 😄 
What about this: "Generate the UCD's shiplist records (known as "container records" in UCD documentation) related to build outputs in partitioned datasets" ?

Reply:
I like your proposal and have updated the README with the recent added commit .